gridcomps/ExtData2G/ExtDataDerived.F90

```diff
@@ -51,6 +55,57 @@ function new_ExtDataDerived(config,unusable,rc) result(rule)
       _RETURN(_SUCCESS)
    end function new_ExtDataDerived
 
+   function get_variables_in_expression(this,rc) result(variables_in_expression)
```

Comment:
I think it might be good to work on refactoring this long procedure before we move on.   Just try to make it as simple as possible.  I could work with you on it, though I don't immediately see specific things I want done.

Reply:
It doesn't seem that long, the only thing I would do is perhaps moving all the masking procedures to a separate class.

Reply:
The masking is a bit of a mess as this was refactor of the old Extdata rather than a clean slate... so the new one just inherited that. I'll work on abstracting the masking into a separate class although might not happen by the end of today (4/12).

Reply:
I moved all the masking to it's own class, should be cleaner.